### PR TITLE
Add docs for connecting profiles and traces

### PR DESCRIPTION
**Pull request by kayayarai**
### What does this PR do?
Recreates PR for adding docs for Code Hotspots

### Motivation
Closing of kari/code-hotspots and PR 9615
DOCS-1477

Preview
https://docs-staging.datadoghq.com/kari/code-hotspots2/tracing/profiler/connect_traces_and_profiles/
https://docs-staging.datadoghq.com/kari/code-hotspots2/tracing/profiler/

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Review by GaurabAryal — APPROVED** [review]

**Comment by kayayarai** [comment]
Created https://datadoghq.atlassian.net/browse/DOCS-1663 to track Documentation team editorial review

**Comment by GaurabAryal** [comment]
@kayayarai This doc is ready to be merged FYI :)

File: content/en/tracing/profiler/connect_traces_and_profiles.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      text: 'APM Distributed Tracing'
+    - link: 'tracing/profiler/getting_started'
+      tag: 'Documentation'
+      text: 'Enable Continuous Profiler for Your Application.'
+    - link: 'tracing/profiler/intro_to_profiling'
+      tag: 'Documentation'
+      text: 'Introduction to Profiling'
+---
+
+If your application is showing performance problems in production, it's helpful to connect the distributed trace information from APM with the full profile of the code stack. Application processes that have both APM distributed tracing and continuous profiler enabled are automatically linked, so you can move directly from span information to profiling data on the Code Hotspots tab, and find specific lines of code related to performance issues.
```
Comment (apigirl) [inline content/en/tracing/profiler/connect_traces_and_profiles.md:16]:
```suggestion
If your application is showing performance problems in production, it's helpful to connect the distributed trace information from APM with the full profile of the code stack. Application processes that have both APM distributed tracing and Continuous Profiler enabled are automatically linked. So you can move directly from span information to profiling data on the Code Hotspots tab and find specific lines of code related to performance issues.
```
this sentence is a lot, but feel free to ignore this suggestion

File: content/en/tracing/profiler/connect_traces_and_profiles.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+{{< img src="tracing/profiling/code_hotspots_tab.gif" alt="Code Hotspots tab shows profiling information for a APM trace span">}}
+
+## Prerequisites
+
+{{< programming-lang-wrapper langs="java,python" >}}
+{{< programming-lang lang="java" >}}
+Code Hotspots identification is enabled by default when you [turn on Profiling for your service][1]. For manually instrumented code, Continuous Profiler requires Scope activation of spans:
```
Comment (apigirl) [inline content/en/tracing/profiler/connect_traces_and_profiles.md:24]: is continuous profiler capitalized? you have it lowercase above

Comment (apigirl) [reply content/en/tracing/profiler/connect_traces_and_profiles.md:24]: I don't think scope is capitalized either

File: content/en/tracing/profiler/connect_traces_and_profiles.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+- **File I/O** shows the time taken waiting for a disk read/write operation to execute.
+- **Socket I/O** shows the time taken waiting for a network read/write operation to execute.
+- **Object wait** shows the time waiting for a notify call on an object.
+- **Other** shows the time taken to execute the span that cannot be explained by profiling data.
+
+Click on one of these types to see a corresponding list, ordered by time, of the methods that are taking time. Clicking on the plus `+` will expand the stack trace to that method **in reverse order**.
+
+It is not uncommon to have a small amount of **Other** unexplained time (less than 10%). Potential reasons for Other time include:
+
+  - The span you selected isn't directly mapped to any execution. Profiling data is associated uniquely to spans when they are executing on a specific thread. For example, some spans are created/used uniquely as virtual containers of a series of related processing steps and never actually directly associated with any thread execution.
```
Comment (apigirl) [inline content/en/tracing/profiler/connect_traces_and_profiles.md:73]:
```suggestion
  - The span you selected isn't directly mapped to any execution. Profiling data is associated uniquely to spans when they are executing on a specific thread. For example, some spans are created and used uniquely as virtual containers of a series of related processing steps and never directly associated with any thread execution.
```

File: content/en/tracing/profiler/connect_traces_and_profiles.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  - Your application process cannot access CPU resources to execute and is paused. There is no way for profiler to know about competing resources from other processes or containers.
```
Comment (apigirl) [inline content/en/tracing/profiler/connect_traces_and_profiles.md:74]:
```suggestion
  - Your application process cannot access CPU resources to execute and is paused. There is no way for the Profiler to know about competing resources from other processes or containers.
```

File: content/en/tracing/profiler/connect_traces_and_profiles.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  - The application is locked in synchronization or in I/O events that are individually lower than 10ms: Java profiler gets data for paused thread events (locks, I/O, parks) that are larger than 10ms. If you want to reduce that threshold, see [the documentation for changing setup defaults][1].
```
Comment (apigirl) [inline content/en/tracing/profiler/connect_traces_and_profiles.md:75]:
```suggestion
  - The application is locked in synchronization or in I/O events that are individually lower than 10ms: the Java Profiler receives data for paused thread events (locks, I/O, parks) that are larger than 10ms. If you want to reduce that threshold, see [the documentation for changing setup defaults][1].
```

File: content/en/tracing/profiler/connect_traces_and_profiles.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+  - The span you selected is short. Profiling is a sampling mechanism that regularly looks at how your code behaves. There might not be enough representative data for spans shorter than 50ms
+  - Missing instrumentation: Profiling breakdown requires that spans are associated with executing threads by activating these spans in the ScopeManager. Some custom instrumentations don't activate these spans properly and so we can't map them to executing threads. If this span comes from a custom integration, see the [Custom Instrumentation docs][2] for information on how to improve this.
```
Comment (apigirl) [inline content/en/tracing/profiler/connect_traces_and_profiles.md:77]:
```suggestion
  - Missing instrumentation: Profiling breakdown requires that spans are associated with executing threads by activating these spans in the ScopeManager. Some custom instrumentations don't activate these spans properly, so we can't map them to executing threads. If this span comes from a custom integration, see the [Custom Instrumentation docs][2] for information on how to improve this.
```

File: content/en/tracing/profiler/connect_traces_and_profiles.md (hunk 2 above)
Comment (kayayarai) [reply content/en/tracing/profiler/connect_traces_and_profiles.md:24]: Ah, yes, I checked with other profiler docs, and am going to go all lower-case, thank you.

**Review by apigirl — APPROVED** [review]

**Pull request merged**